packages/styled/components/image/irvingImg.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import useIntersect from 'hooks/useIntersect';
+import { Image } from './styles';
+
+const IrvingImg = (props) => {
+  const {
+    alt,
+    aspectRatio,
+    sizes,
+    src,
+    srcset,
+    onError,
+    lazyload,
+    lqipSrc,
+  } = props;
+  const [currentSrc, setCurrentSrc] = useState(lqipSrc);
+  const [currentSrcSet, setCurrentSrcSet] = useState(null);
+  const [setNode, entry] = useIntersect();
+  const hasSrc = (currentSrc && currentSrc !== lqipSrc);
+  // Set src and srcset if entry matches current node and src isn't already set.
+  useEffect(() => {
+    if (
+      entry.target &&
+      0 < entry.intersectionRatio &&
+      ! hasSrc
+    ) {
+      setCurrentSrc(src);
+      setCurrentSrcSet(srcset);
+    }
+  }, [entry]);
+
+  return (
+    <Image
+      alt={alt}
+      aspectRatio={aspectRatio}
+      sizes={sizes}
+      src={currentSrc}
+      srcSet={currentSrcSet}
+      onError={onError}
+      lazyload={lazyload}
+      ref={setNode}
+      hasSrc={hasSrc}
+    />
+  );
+};
+
+IrvingImg.propTypes = {
+  aspectRatio: PropTypes.oneOfType([
```

Comment:
The proptypes of `Image` are documented, should these be documented as well?

Reply:
They're a subset of the same parent image component props, so figured I'd just leave them off for documentation.